reflex/app.py

```diff
@@ -404,11 +404,14 @@ def _check_routes_conflict(self, new_route: str):
 
     def add_custom_404_page(
         self,
-        component,
-        title=None,
-        image=None,
-        description=None,
-        meta=constants.DEFAULT_META_LIST,
+        component: Optional[Union[Component, ComponentCallable]] = None,
```

Comment:
Why is the component optional if they're adding a page?

Reply:
If we setup a redirect, the actual page to render doesn't matter (and will be set to default as a rx.fragment()